src/Matrix.ts

```diff
@@ -801,6 +801,41 @@ export default class Matrix extends NDArray {
 
     return result;
   }
+
+  /**
+   * Adds a multiple of one row to another inside a matrix.
+   * INPUT: 
+   *        - dest (destination) is the row for adding the other 
+   *          row that was multiplied with the scalar.
+   *        - source : is the row that would be multiplied with 
+   *          the scalar (for adding)
+   *        - scalar : the scalar of type number.
+   * OUTPUT: The current changed matrix.
+   */
+  rowAdd(dest : number, source : number, scalar : number = 1.0 ) : Matrix {
+
+    // preconditions
+    this.check(dest, 0);
+    this.check(source, 0);
+    if (typeof scalar != "number")
```

Comment:
There are no runtime type checks in the other methods, no reason to do it here.

Reply:
@christianbender Sorry, I only meant the `if (typeof scalar != "number")`! The out of bounds checks (`this.check(dest, 0)` and `this.check(source, 0)`) should definitely still be there :)